src/ape/api/providers.py

```diff
@@ -136,10 +136,24 @@ def decode(cls, data: dict) -> "ReceiptAPI":
         ...
 
 
+@abstractdataclass
+class BlockGasAPI:
+    raw_data: Dict
+
+    @property
+    @abstractmethod
+    def base_fee(self) -> Optional[int]:
+        ...
+
+    @property
+    @abstractmethod
+    def total_gas_used(self) -> int:
+        ...
+
+
 @abstractdataclass
 class BlockAPI:
-    gas_fee_data: Dict
-    consensus_data: Dict
```

Comment:
deleted `consensus_data`

I definitely don't see using this data a whole lot, except for some rare queries made against the protocol itself

Reply:
Yeah I figured we could re-add it later when we had a better idea of what the API should be? I just am unsure about that one, as far as making higher level goes.

Reply:
just `block.consensus["difficulty"]` maybe

Reply:
I added the class for it so we can add to it and made sure difficulty was in there from `ape_ethereum` (provided it exists, its optional).